nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentModelJsonWriter.java

```diff
@@ -258,4 +270,24 @@ private void writeSchemaProperties(JsonGenerator jg, DocumentModel doc, String s
         }
     }
 
+    protected void withDocumentAttached(DocumentModel doc, ThrowableRunnable<IOException> runnable) throws IOException {
+        if (doc.getSessionId() == null) {
```

Comment:
This check is already done by the caller. Or did you want to move all checks inside `withDocumentAttached` to avoid complexity in the caller?

Reply:
Good catch. And yes, I want to let the checks on the caller.